be/src/exec/pipeline/exchange/exchange_sink_operator.cpp

```diff
@@ -659,17 +664,25 @@ Status ExchangeSinkOperator::serialize_chunk(const vectorized::Chunk* src, Chunk
     return Status::OK();
 }
 
-void ExchangeSinkOperator::construct_brpc_attachment(PTransmitChunkParamsPtr chunk_request, butil::IOBuf& attachment) {
+int64_t ExchangeSinkOperator::construct_brpc_attachment(PTransmitChunkParamsPtr chunk_request,
+                                                        butil::IOBuf& attachment) {
+    int64_t attachment_physical_bytes = 0;
     for (int i = 0; i < chunk_request->chunks().size(); ++i) {
         auto chunk = chunk_request->mutable_chunks(i);
         chunk->set_data_size(chunk->data().size());
+
+        int64_t before_bytes = CurrentThread::current().get_consumed_bytes();
         attachment.append(chunk->data());
```

Comment:
Do not consider the `clear_data()` and `shrink_to_fit()` ?

Reply:
Needn't consider them, because we only need record the memory of serialized `attachment`, while `chunk` is always released in the driver worker threads.